autofs/connectathon/cthon_automount/bin/test.proto_check

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+#
+#ident	"$Id: proto_check,v 1.1 2005/10/25 22:38:34 jmoyer Exp $"
+#
+# checks that the expected transport has been used
+# for the NFS mount.
+#
+# This test is heavily dependent on the output
+# of the nfsstat -m command. If and when the output
+# changes this test will likely need to be revised.
+#
+
+. ../src/func.common
+
+check_proto() {
+	Proto=$1
+	Ln=`nawk -F\: '{print $1}' /tmp/grepn.out.$$`
+	Ln=`expr $Ln + 1`
+	head -$Ln /tmp/nfsstat.out.$$ | tail -1 > /tmp/proto.out.$$
+	Po=`nawk '{print $2}' /tmp/proto.out.$$`
+	echo "\tExpect proto=${Proto}, got $Po"
+	Found=`try_to_grock "proto=$Proto" "$Po"`
+	# Consider <proto>6 equivalent to <proto> from protocol POV.
+	if [ -z "$Found" ]; then
+		Found=`try_to_grock "proto=${Proto}6" "$Po"`
+	fi
+	if [ "$Found" != "found" ]; then
+		echo "\tFAILED"
+		fail=`expr ${fail} + 1`
+	else
+		echo "\tPASSED."
+	fi
+}
+	
+do_nfsstat() {
+	if [ x${VERBOSE} != x ]; then
+		echo "\texamine nfsstat -m output..."
+	fi
+	nfsstat -m > /tmp/nfsstat.out.$$
+
+	grep -n "^$1" /tmp/nfsstat.out.$$ > /tmp/grepn.out.$$ 2>&1
+	if [ $? -ne 0 ]; then
+		echo "$1 not found in nfsstat, exiting."
+		exit 1
```

Comment:
Seems like another setup issue ? I would review all the places it returns 1 and add the cki_lib function to abort the task, we want to avoid failing the kernel for a setup issue.

Reply:
@ftjahjad  what about this comment ? Should we abort the task ?